Roachtest redirect SSH flakes to test-eng

smg260 opened the pull request:
*See second commit note at the bottom* 

This PR inspects the failure output of a roachtest, and if it sees an SSH_PROBLEM, overrides the owning team to test-eng when reporting the github issue. 

Currently errors are classified as an `SSH` error by roachprod if the exit code is `255` with an accompanying message prefixed with `SSH_PROBLEM` [[1]](https://github.com/cockroachdb/cockroach/blob/ad3bd1355463cefdc07e995765fa82adfe391d05/pkg/roachprod/errors/errors.go#L112). The errors are stringified and saved into `t.mu.output|failureMsg`. Thus in the test_runner at the call site of issue posting, we can check `t.mu.output` for `SSH_PROBLEM` and override the team and issue name accordingly.


Resolves: https://github.com/cockroachdb/cockroach/issues/82398

Release justification: test-only change
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/88492)
<!-- Reviewable:end -->

tbg commented:
We can also build on top of https://github.com/cockroachdb/cockroach/pull/88556 for this, I have a DM going with Miral.

tbg commented:
@smg260 (correct me if I'm wrong please) is getting https://github.com/cockroachdb/cockroach/pull/88556 in first and then re-working this to build on top of that PR.

I'm unassigning for review so that it doesn't show up in my queue in the meantime.

smg260 commented:
@tbg Yep. Marking as draft for now.

Commit:
roachtest: Reassign roachtest failures which may have been caused by
 SSH flakes to the test-eng team.

 Also abstracts away how issue name and team is overriden based on
 an issue category, which can be ClusterCreation, SSH, or Other

 Resolves issue #82398

 Release justification: test-only change
 Release note: None

smg260 commented:
This is rebased on top of https://github.com/cockroachdb/cockroach/pull/88556 which introduces structured errors. 

Detecting the SSH flake via substring still works and keeps this PR simple. 

This might be a good place to introduce some retry logic for running the SSH commands

srosenberg commented:
> Detecting the SSH flake via substring still works and keeps this PR simple.

A chance of a false positive (via substring) is probably very low, but structured errors should make it cleaner and pave the way for more cases. I am thinking of a switch statement on something like `errCategory` with cluster creation and ssh errors injecting the corresponding category at the site?

> 
> This might be a good place to introduce some retry logic for running the SSH commands

There is a separate issue to deal with SSH retries: https://github.com/cockroachdb/cockroach/issues/73542

srosenberg commented:
I recently spotted another issue... seeing this error message quite often during cluster teardown,

```
teardown: 16:53:25 cluster.go:1149: failed to fetch logs: cluster.Get: get logs failed
```

The above error message originates in [1]. The problem is that the actual error message is swallowed [2] because `l.File` is usually non-nil, when invoked from `roachtest`. (The logger is essentially the root logger and `l.File` is `test.log`.)  l believe the same issue applies to `Put`.

[1] https://github.com/cockroachdb/cockroach/blob/master/pkg/roachprod/install/cluster_synced.go#L2034
[2] https://github.com/cockroachdb/cockroach/blob/master/pkg/roachprod/install/cluster_synced.go#L2028

renatolabs commented on pkg/cmd/roachtest/github.go:
```diff
@@ -32,6 +32,14 @@ type githubIssues struct {
+type issueCategory int
+
+const (
+	clusterCreation issueCategory = iota
```
Nit: these are very generic names that will be shared across all roachtest files (as they are all `package main`). We could add an `Error` suffix to these (`sshError`, `otherError`, etc).

renatolabs commented on pkg/cmd/roachtest/github.go:
```diff
@@ -80,7 +102,7 @@ func (g *githubIssues) createPostRequest(t test.Test, message string) issues.Pos
 	}
 
-	artifacts := fmt.Sprintf("/%s", t.Name())
+	artifacts := fmt.Sprintf("/%s", issueName)
```
I think this is going to break the artifacts link on the GitHub issues. The artifacts path will still be under the original test's name.

smg260 replied:
that's a good catch - it would have been breaking on master too but we only override for cluster creation there

renatolabs commented on pkg/cmd/roachtest/github.go:
```diff
@@ -112,7 +134,7 @@ func (g *githubIssues) createPostRequest(t test.Test, message string) issues.Pos
 		ProjectColumnID: projColID,
 		PackageName:     "roachtest",
-		TestName:        t.Name(),
+		TestName:        issueName,
```
I wish we didn't have to do this. Ideally, I'd want the test failure to be: `roachtest: TestFoo failed (ssh_problem)` instead of the cryptic `roachtest: ssh_problem failed`. We will have to go through the artifacts each time to find out which test actually failed.

I'd suggest we don't overwrite the test name, except that for the `cluster_creation` error, if the cloud provider is having issues, typically lots of test will fail and we don't want to create a separate issue for each one.

smg260 replied:
It probably makes sense to keep SSH flakes together, instead of having a new one created for each incidence. 

We can prepend the test name to the message body like cluster creation does?

renatolabs reviewed (commented):
> structured errors should make it cleaner and pave the way for more cases

I agree with this, I think structured errors open the door for us to do things based on the error types. I think that's cleaner and also wouldn't break if someone changed how `errors.SSH` is formatted.

As general food for thought for everyone: IMO, errors like cluster creation and SSH problems are better reported in a more concise way other than a GitHub issue. We just want to be notified about them, but there's little (nothing?) we can do about them other than acknowledge and close the issue when it's opened.

One idea is to post something on Slack (in the existing test-eng-ops) channel with a summary of a roachtest nightly run, including how many tests were skipped due to cluster creation errors, SSH flakes, etc.

smg260 commented:
> A chance of a false positive (via substring) is probably very low, but structured errors should make it cleaner and pave the way for more cases. I am thinking of a switch statement on something like `errCategory` with cluster creation and ssh errors injecting the corresponding category at the site?

That's definitely the goal, and my thoughts were to introduce that when we switch to `cockroachdb/errors`. We'd then be able to annotate errors using something like `errors.Mark(err, SSHerr)` or `errors.Mark(err, ClusterCreationErr) at the point we detect it. (tbg's idea).

So we can keep this PR tight in scope and address just the SSH flakes, or expand it to more generically handle different error types. The PR does at least currently constrain the logic to to determine the owning team in `githib.go` (as it should). 


> There is a separate issue to deal with SSH retries: #73542

👍 This was getting a bit gnarly to implement in my test branch

smg260 commented:
>As general food for thought for everyone: IMO, errors like cluster creation and SSH problems are better reported in a more concise way other than a GitHub issue. We just want to be notified about them, but there's little (nothing?) we can do about them other than acknowledge and close the issue when it's opened.
> 
> One idea is to post something on Slack (in the existing test-eng-ops) channel with a summary of a roachtest nightly run, including how many tests were skipped due to cluster creation errors, SSH flakes, etc.

I agree that some errors are better reported via other mediums such as Slack, but having the same error reported a shared github issue (or similar) provides a better system of record. Ultimately, where an issue is posting should be agnostic to the caller and the posting code could drive which channels are notified via some coniguration

Commit:
roachtest: Utilise structured errors for detection of SSH flakes.

This commit utilises cockroachdb/errors markers api to mark an SSH
error with exit code of 255. The test runner, when it is posting an
issue to github, can act according to which marker may be present
in t.mu.failures[0]. In this case, we override the owning team to
protect them from having to investigate what is likely a transient
SSH issue.

A test now has the concept of a failure[] instead of error[]. Each
failure contains the original `squashedErr` and an errors[],
which are all the errors passed via t.Failf/Errorf.

We can then preserve the relationship of multiple
errors to any particular failure within a test, and match on any given
reference error (like SSH flakes from above)

This moves us away from substring matching on error messages.

The test name is prepended to both cluster creation and ssh errors
to ease troubleshooting as all those issues will be bucketed
under either category respectively.

Release justification: test-only change
Release note: None

smg260 commented:
The second commit switches to using errors/marker api to mark an SSH flake where it is detected in roachprod. This allows us to look for a marker exception and act on it accordingly.

Summary of changes:

1. switch from `[]error` to `[]failure` in `t.mu` where `failure` is a struct containing all errors which were passed as args. The impetus for this was to preserve secondary+ exceptions which are passed into the `t.Error/Fatal` methods. Currently when `t.Error(someErr)` is invoked, a new error is created, moving `someErr` to a `secondary` error, out of the primary error chain. This makes is very hard to search for the marker which would have been inserted at `someErr` but is now *not* visible to `errors.Is(..)`
2. Add `ErrSSH255` and `errClusterProvisioningFailed` reference errors (the marker errors). We control what we post in the issue according to if we find these
3. remove unused members of `t.mu`
4. remove `t.argsToErr` in favour of `errors.NewWithDepthf`

tbg reviewed (commented):
> Reviewed 4 of 4 files at r2, 7 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88492)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @renatolabs, @smg260, and @srosenberg)

___
*[`pkg/cmd/roachtest/github.go` line 137 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NDYOj-KltlIVTb_NwwB-r1-137:-NE_5VZd74QllHgcmtz3:b-esdt41) ([raw file](https://github.com/cockroachdb/cockroach/blob/70db1029fba8e7276cf3bbf8b6fd2869f26bd868/pkg/cmd/roachtest/github.go#L137)):*
<details><summary><i>Previously, smg260 (Miral Gadani) wrote…</i></summary><blockquote>

It probably makes sense to keep SSH flakes together, instead of having a new one created for each incidence. 

We can prepend the test name to the message body like cluster creation does?
</blockquote></details>

I agree that this isn't excessively pretty, but will it maybe get the job done just fine? For example, the cluster_creation messages mention the test name sufficiently prominently:

https://github.com/cockroachdb/cockroach/issues/89810

Not sure if this is the case here too (since there won't be a `.Skip()`) but the message also contains the test name right at the top: https://github.com/cockroachdb/cockroach/issues/90045#issue-1411156917

We can also go through the refactor and add a title override that we can use instead of the test name. I'm just not sure it's worth the time.
___
*[`pkg/cmd/roachtest/github.go` line 40 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_5HuG8h_pmURTpVEV:-NE_5HuH1aI4WLKUludF:b-18jrr8) ([raw file](https://github.com/cockroachdb/cockroach/blob/0000095bee8f09fe4ac0e5704799a40205024f8e/pkg/cmd/roachtest/github.go#L40)):*
> ```Go
> 	clusterCreationErr issueCategory = iota
> 	sshErr
> 	otherErr
> ```

tiny nit: since otherErr is the catch-all, it should be represented by the zero value.
___
*[`pkg/cmd/roachtest/github.go` line 162 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_72xo1DUe_3P-Ki7Q:-NE_72xo1DUe_3P-Ki7R:b-ffi5u) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/cmd/roachtest/github.go#L162)):*
> ```Go
> 
> 	//TODO: perhaps remove category completely and move this to
> 	// issuePoster?
> ```

The poster has multiple callers and so I think this should continue to live in roachtest.
___
*[`pkg/cmd/roachtest/test_impl.go` line 45 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_7EzWD2H8dJHdfQ32:-NE_7EzWD2H8dJHdfQ33:b-e935w1) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/cmd/roachtest/test_impl.go#L45)):*
> ```Go
> 	// errors are all the errors passed to a single invokation of
> 	// `addFailure`
> 	errors []error
> ```

Add some color here, why are we stashing these errors? It's because they might be structured and we want to preserve that (which `fmt.Sprintf(...)` wouldn't do). Maybe given an example that `t.Fatalf("foo %s %s %s", "hello", err1, err2)` would lead to `[]error{err1,err2}`.
___
*[`pkg/cmd/roachtest/test_impl.go` line 280 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_7T-63dvfAuocbNY-:-NE_7T-63dvfAuocbNY0:b-mn3fzy) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/cmd/roachtest/test_impl.go#L280)):*
> ```Go
> // from a test's closure. The test runner itself should never call this.
> func (t *testImpl) Fatal(args ...interface{}) {
> 	t.Fatalf("%v", args...)
> ```

This will print `t.Fatal("hi")` as `["hi"]` which isn't what we want. I think you want something like

```go
t.Fatalf(strings.Join(strings.Repeat(len(args), " %v"))[1:], args...)
```
___
*[`pkg/cmd/roachtest/test_impl.go` line 287 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_7w8t50cUzOCKSrVK:-NE_7w8t50cUzOCKSrVL:b9mrpj0) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/cmd/roachtest/test_impl.go#L287)):*
> ```Go
> 	errs := []error{errors.NewWithDepthf(1, format, args...)}
> 	errs = append(errs, collectErrors(args...)...)
> 	t.addFailure(newFailure(errs...))
> ```

The symmetry in `newFailure` is a little misleading then since the first error is special, how about

```go
type failure struct {
  squashedErr error
  wrapped []error
}
```

or something like that?
___
*[`pkg/cmd/roachtest/test_impl.go` line 298 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_7lgB0ZoCIzEgyiTh:-NE_7lgB0ZoCIzEgyiTi:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/cmd/roachtest/test_impl.go#L298)):*
> ```Go
> 
> func (t *testImpl) Error(args ...interface{}) {
> 	t.Errorf("%v", args...)
> ```

ditto
___
*[`pkg/roachprod/errors/errors.go` line 123 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_8DUyBcBPLGmGvgbl:-NE_8DUyBcBPLGmGvgbm:bxv2gqe) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/roachprod/errors/errors.go#L123)):*
> ```Go
> 	if exitErr, ok := asExitError(err); ok {
> 		if exitErr.ExitCode() == 255 {
> 			return SSH{errors.Mark(err, ErrSSH255)}
> ```

Does this work? Does `SSH{}` implement the necessary `Unwrap()` method?


<!-- Sent from Reviewable.io -->

smg260 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88492)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @renatolabs, @srosenberg, and @tbg)

___
*[`pkg/cmd/roachtest/github.go` line 105 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NDYOj-KltlIVTb_NwwB-r1-105:-NEb0b-5EZYmBGZA45N8:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/70db1029fba8e7276cf3bbf8b6fd2869f26bd868/pkg/cmd/roachtest/github.go#L105)):*
<details><summary><i>Previously, smg260 (Miral Gadani) wrote…</i></summary><blockquote>

that's a good catch - it would have been breaking on master too but we only override for cluster creation there
</blockquote></details>

Done
___
*[`pkg/cmd/roachtest/github.go` line 162 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_72xo1DUe_3P-Ki7Q:-NEb0hb2CbM9oxSpQkmg:b-81vgec) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/cmd/roachtest/github.go#L162)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

The poster has multiple callers and so I think this should continue to live in roachtest.
</blockquote></details>

Yep sorry - unclear TODO. Was a thought to myself to remove the idea of a category, and instead depend completely on reference errors. `issuePoster` here refers to the `createPostRequest` function in roachtest/github.go
___
*[`pkg/cmd/roachtest/test_impl.go` line 280 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_7T-63dvfAuocbNY-:-NEb16LD-RsWs5pPGjgM:bpgjcto) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/cmd/roachtest/test_impl.go#L280)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

This will print `t.Fatal("hi")` as `["hi"]` which isn't what we want. I think you want something like

```go
t.Fatalf(strings.Join(strings.Repeat(len(args), " %v"))[1:], args...)
```
</blockquote></details>

Done. Thank you. Got lost in the variadic args being passed from func to func. 
___
*[`pkg/cmd/roachtest/test_impl.go` line 287 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_7w8t50cUzOCKSrVK:-NEb1HTb4LnyC2EBpdyL:b-u54w2x) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/cmd/roachtest/test_impl.go#L287)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

The symmetry in `newFailure` is a little misleading then since the first error is special, how about

```go
type failure struct {
  squashedErr error
  wrapped []error
}
```

or something like that?
</blockquote></details>

Agreed. That is much better.
___
*[`pkg/roachprod/errors/errors.go` line 123 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NE_8DUyBcBPLGmGvgbl:-NEb1RME9rqDu04_9w60:bo0eyja) ([raw file](https://github.com/cockroachdb/cockroach/blob/7beb8a916082633b1abf942c063b4ef1ab0a36aa/pkg/roachprod/errors/errors.go#L123)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Does this work? Does `SSH{}` implement the necessary `Unwrap()` method?
</blockquote></details>

Yep .  In `roachprod/errors.go`



_Code snippet:_
```Go
func (e SSH) Unwrap() error {
	return e.Err
}
```


<!-- Sent from Reviewable.io -->

Commit:
roachtest: Set localhost as the http listener when --local flag
is used. This prevents OSX from prompting to allow incoming
network connections when running a roachtest from an ide or
 shell.

Release justification: test-only change
Release note: None

tbg reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-:-NF8VHcVBwGM6m9SFfah:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @renatolabs, @smg260, and @srosenberg)

___
*[`-- commits` line 39 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8VFKD5jiRianFO2VZ):*
:love_letter: 
___
*[`pkg/cmd/roachtest/test_impl.go` line 48 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8UsHL78iLufKrqkA5:-NF8UsHL78iLufKrqkA6:b-c8wd61) ([raw file](https://github.com/cockroachdb/cockroach/blob/921c15338178c5c4711e492f715efcf6664e046e/pkg/cmd/roachtest/test_impl.go#L48)):*
> ```Go
> // being captured in the squashedErr
> type failure struct {
> 	// this is the single error created from variadic args passed to t.Error(f)/t.Fatal(f)
> ```

general comment, our style guide asks for complete sentences incl punctuation and capitalization (except for inline comments where it's the opposite), i.e.

// This is the single error created from variadic args passed to `t.{Fatal,Error}{,f}`.

I usually don't point out isolated incidents of this but it looks like it's happening throughout here, it will save you lots of nits down the road to get the muscle memory right. :-)
___
*[`pkg/cmd/roachtest/test_impl.go` line 286 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8RvHH8nKpG-spO511:-NF8RvHH8nKpG-spO512:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/921c15338178c5c4711e492f715efcf6664e046e/pkg/cmd/roachtest/test_impl.go#L286)):*
> ```Go
> // from a test's closure. The test runner itself should never call this.
> func (t *testImpl) Fatal(args ...interface{}) {
> 	t.Error(args...)
> ```

ditto
___
*[`pkg/cmd/roachtest/test_impl.go` line 292 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8LDdCBuMLCkJ1fr2_:-NF8LDdCBuMLCkJ1fr2a:b914pzh) ([raw file](https://github.com/cockroachdb/cockroach/blob/921c15338178c5c4711e492f715efcf6664e046e/pkg/cmd/roachtest/test_impl.go#L292)):*
> ```Go
> // Fatalf is like Fatal, but takes a format string.
> func (t *testImpl) Fatalf(format string, args ...interface{}) {
> 	t.Errorf(format, args...)
> ```

Yo zneed to call `addFailure` because you'l 
___
*[`pkg/cmd/roachtest/test_impl.go` line 298 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8L-TC8fQ_MCPbuGgc:-NF8L-TC8fQ_MCPbuGgd:bkmsbf) ([raw file](https://github.com/cockroachdb/cockroach/blob/921c15338178c5c4711e492f715efcf6664e046e/pkg/cmd/roachtest/test_impl.go#L298)):*
> ```Go
> // FailNow implements the TestingT interface.
> func (t *testImpl) FailNow() {
> 	t.addFailure("FailNow called", nil)
> ```

passing `nil` here doesn't seem right, `FailNow called` doesn't have a verb in it and besides a `nil` doesn't add any info.
___
*[`pkg/cmd/roachtest/test_impl.go` line 400 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8Kopb2MRcGaMIZXDa:-NF8Kopb2MRcGaMIZXDb:b-7yxro8) ([raw file](https://github.com/cockroachdb/cockroach/blob/921c15338178c5c4711e492f715efcf6664e046e/pkg/cmd/roachtest/test_impl.go#L400)):*
> ```Go
> // failureContainsError returns true if any of the errors in a given failure
> // matches the reference error
> func failureContainsError(refError error, f failure) bool {
> ```

nit: switch the order of the arguments, to reflect "failure contains error", not "error contains failure".
___
*[`pkg/testutils/lint/passes/fmtsafe/functions.go` line 96 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8KL-BBMhDohLzJtGm:-NF8KL-BBMhDohLzJtGn:b-ad6j83) ([raw file](https://github.com/cockroachdb/cockroach/blob/921c15338178c5c4711e492f715efcf6664e046e/pkg/testutils/lint/passes/fmtsafe/functions.go#L96)):*
> ```Go
> 	// Both of these signatures need to be included for the linter to not flag
> 	// roachtest testImpl.addFailure since it is in the main package
> 	// This could be a bug in nogo
> ```

Did you mean to leave this? If so, mind linking an issue?


<!-- Sent from Reviewable.io -->

smg260 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88492)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @renatolabs, @srosenberg, and @tbg)

___
*[`pkg/cmd/roachtest/test_impl.go` line 48 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8UsHL78iLufKrqkA5:-NF9Qtd64oynyf2xmqZt:b-qhtcwb) ([raw file](https://github.com/cockroachdb/cockroach/blob/921c15338178c5c4711e492f715efcf6664e046e/pkg/cmd/roachtest/test_impl.go#L48)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

general comment, our style guide asks for complete sentences incl punctuation and capitalization (except for inline comments where it's the opposite), i.e.

// This is the single error created from variadic args passed to `t.{Fatal,Error}{,f}`.

I usually don't point out isolated incidents of this but it looks like it's happening throughout here, it will save you lots of nits down the road to get the muscle memory right. :-)
</blockquote></details>

:+1:  Thanks, missed the capitalisation. I don't think it's happened anywhere else though
___
*[`pkg/cmd/roachtest/test_impl.go` line 286 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8RvHH8nKpG-spO511:-NF9R8Ni-FhvxvX64rvA:by4qanq) ([raw file](https://github.com/cockroachdb/cockroach/blob/921c15338178c5c4711e492f715efcf6664e046e/pkg/cmd/roachtest/test_impl.go#L286)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

What is missing here?
___
*[`pkg/cmd/roachtest/test_impl.go` line 292 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88492#-NF8LDdCBuMLCkJ1fr2_:-NF9YuqlCwT0TDw868sF:bhmkuq9) ([raw file](https://github.com/cockroachdb/cockroach/blob/921c15338178c5c4711e492f715efcf6664e046e/pkg/cmd/roachtest/test_impl.go#L292)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Yo zneed to call `addFailure` because you'l 
</blockquote></details>

Cut off comment, but `addFailure` is being called via `t.Errorf`


<!-- Sent from Reviewable.io -->

smg260 commented:
Test name will be prepended to issue body in the case of cluster creation AND ssh flakes. 

Other conversations resolved. Will merge after TC builds are green.

smg260 commented:
bors r=tbg

smg260 commented:
bors retry
